pkg/controller/gittrack/gittrack_controller_test.go

```diff
@@ -324,62 +384,75 @@ var _ = Describe("GitTrack Suite", func() {
 	})
 })
 
-var getsFilesFromRepo = func(path string, count int) {
-	Context(fmt.Sprintf("With subPath %s", path), func() {
-		var files map[string]*gitstore.File
-		var gt *farosv1alpha1.GitTrack
+var _ = Describe("ClusterGitTrack Suite", func() {
+	var c client.Client
+	var mgr manager.Manager
+	var instance farosv1alpha1.GitTrackInterface
+	var requests chan reconcile.Request
+	var stop chan struct{}
+	var r reconcile.Reconciler
 
-		BeforeEach(func() {
-			var err error
-			var reconciler *ReconcileGitTrack
-			var ok bool
-			reconciler, ok = r.(*ReconcileGitTrack)
-			Expect(ok).To(BeTrue())
-			gt = &farosv1alpha1.GitTrack{
-				ObjectMeta: metav1.ObjectMeta{
-					Name:      "test",
-					Namespace: "default",
-				},
-				Spec: farosv1alpha1.GitTrackSpec{
-					SubPath:    path,
-					Repository: repositoryURL,
-					DeployKey:  farosv1alpha1.GitTrackDeployKey{},
-					Reference:  "51798af1c1374d1d375a0eb7a3e53dd67ac5d135",
-				},
-			}
+	const timeout = time.Second * 5
+	const filePathRegexp = "^[a-zA-Z0-9/\\-\\.]*\\.(?:yaml|yml|json)$"
+	const doesNotExistPath = "does-not-exist"
+	const repeatedReference = "448b39a21d285fcb5aa4b718b27a3e13ffc649b3"
 
-			Expect(c.Create(context.TODO(), gt)).NotTo(HaveOccurred())
-			req := reconcile.Request{
-				NamespacedName: types.NamespacedName{
-					Name:      "test",
-					Namespace: "default",
-				},
-			}
-			Eventually(requests, timeout).Should(Receive(Equal(req)))
+	var createInstance = func(gt farosv1alpha1.GitTrackInterface, ref string) {
+		spec := gt.GetSpec()
+		spec.Reference = ref
+		gt.SetSpec(spec)
+		err := c.Create(context.TODO(), gt)
+		Expect(err).NotTo(HaveOccurred())
+	}
 
-			files, err = reconciler.getFiles(gt)
-			Expect(err).ToNot(HaveOccurred())
-		})
+	BeforeEach(func() {
+		farosflags.Namespace = "default"
```

Comment:
This flag isn't relevant to the ClusterGitTracks so if this is how we are going to structure the tests we shouldn't need to set this